nt-list/src/list/base.rs

```diff
@@ -73,47 +73,49 @@ where
         // - The last element of `self` shall be changed to the last element of `other`.
         (*self.blink).flink = other.flink;
         (*other.flink).blink = self.blink;
-        (*other.blink).flink = self.as_ref().end_marker();
+        (*other.blink).flink = self.as_mut().end_marker_mut();
         self.get_unchecked_mut().blink = other.blink;
 
         // Clear `other` without touching any of its elements.
-        let other_end_marker = other.as_ref().end_marker();
-        let other_mut = other.get_unchecked_mut();
-        other_mut.flink = other_end_marker;
-        other_mut.blink = other_end_marker;
+        other.clear();
     }
 
     /// Provides a reference to the last element, or `None` if the list is empty.
     ///
     /// This operation computes in *O*(*1*) time.
     pub unsafe fn back(self: Pin<&Self>) -> Option<&E> {
-        (!self.is_empty()).then(|| (*self.blink).containing_record())
+        (!self.is_empty()).then(|| NtListEntry::containing_record(self.blink))
     }
 
     /// Provides a mutable reference to the last element, or `None` if the list is empty.
     ///
     /// This operation computes in *O*(*1*) time.
     pub unsafe fn back_mut(self: Pin<&mut Self>) -> Option<&mut E> {
-        (!self.as_ref().is_empty()).then(|| (*self.blink).containing_record_mut())
+        (!self.as_ref().is_empty()).then(|| NtListEntry::containing_record_mut(self.blink))
     }
 
     /// Removes all elements from the list.
     ///
     /// This operation computes in *O*(*1*) time, because it only resets the forward and
     /// backward links of the header.
-    pub fn clear(self: Pin<&mut Self>) {
-        let end_marker = self.as_ref().end_marker();
+    pub fn clear(mut self: Pin<&mut Self>) {
+        let end_marker = self.as_mut().end_marker_mut();
         let self_mut = unsafe { self.get_unchecked_mut() };
 
         self_mut.flink = end_marker;
         self_mut.blink = end_marker;
     }
 
-    /// Returns the "end marker element" (which is the address of our own `NtListHead`, but interpreted as a `NtListEntry` element address).
-    pub(crate) fn end_marker(self: Pin<&Self>) -> *mut NtListEntry<E, L> {
+    /// Returns a const pointer to the "end marker element" (which is the address of our own `NtListHead`, but interpreted as a `NtListEntry` element address).
+    pub(crate) fn end_marker(self: Pin<&Self>) -> *const NtListEntry<E, L> {
         (self.get_ref() as *const _ as *mut Self).cast()
```

Comment:
```suggestion
        (self.get_ref() as *const Self).cast()
```

Now that `end_marker()` outputs a `*const NtListEntry`, we don't need another intermediate conversion step, even less a mutable one.